assets/js/modules/reader-revenue-manager/datastore/publications.test.js

```diff
@@ -16,115 +17,202 @@
+
+				// Set expectations for publication ID.
+				expect(
+					registry
+						.select( MODULES_READER_REVENUE_MANAGER )
+						.getPublicationID()
+				).toEqual( 'ABCDEFGH' );
+
+				// Set expectations for publication onboarding state.
+				expect(
+					registry
+						.select( MODULES_READER_REVENUE_MANAGER )
+						.getPublicationOnboardingState()
+				).toEqual( PUBLICATION_ONBOARDING_STATES.UNSPECIFIED );
+
+				const syncTimeMs = registry
 					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				await untilResolved(
-					registry,
-					MODULES_READER_REVENUE_MANAGER
-				).getPublications();
+					.getPublicationOnboardingStateLastSyncedAtMs();
 
-				expect( fetchMock ).not.toHaveFetched( publicationsEndpoint );
-				expect( publications ).toEqual( fixtures.publications );
-				expect( publications ).toHaveLength(
-					fixtures.publications.length
-				);
+				// Ensure that the sync time is set.
+				expect( syncTimeMs ).not.toBe( 0 );
+
+				// Ensure that date is within the last 5 seconds.
+				expect( syncTimeMs ).toBeLessThanOrEqual( Date.now() );
+
+				expect( syncTimeMs ).toBeGreaterThan( Date.now() - 5000 );
 			} );
 
-			it( 'should dispatch an error if the request fails', async () => {
-				const response = {
-					code: 'internal_server_error',
-					message: 'Internal server error',
-					data: { status: 500 },
+			it( 'should set UI_KEY_SHOW_RRM_PUBLICATION_APPROVED_NOTIFICATION to true in CORE_UI store', async () => {
+				registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.receiveGetPublications( fixtures.publications );
+
+				const publication = fixtures.publications[ 3 ];
+
+				// Set the current settings.
+				const settings = {
+					publicationID: publication.publicationId,
+					publicationOnboardingState:
+						PUBLICATION_ONBOARDING_STATES.UNSPECIFIED,
+					publicationOnboardingStateLastSyncedAtMs: 0,
 				};
 
-				fetchMock.getOnce( publicationsEndpoint, {
-					body: response,
-					status: 500,
+				fetchMock.post( settingsEndpoint, {
+					body: {
+						...settings,
+						publicationOnboardingState:
+							PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
+						publicationOnboardingStateLastSyncedAtMs: Date.now(),
+					},
+					status: 200,
 				} );
 
 				registry
-					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				await untilResolved(
-					registry,
-					MODULES_READER_REVENUE_MANAGER
-				).getPublications();
-				expect( fetchMock ).toHaveFetchedTimes( 1 );
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.receiveGetSettings( settings );
 
-				const publications = registry
-					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				expect( publications ).toBeUndefined();
-				expect( console ).toHaveErrored();
+				const uiKeyBeforeAction = registry
+					.select( CORE_UI )
+					.getValue(
+						UI_KEY_SHOW_RRM_PUBLICATION_APPROVED_NOTIFICATION
+					);
```

Comment:
We might as well assert here - it means the test will fail sooner rather than potentially get stuck in an await and not log a failed assertion for this value; it would also be more consistent with other tests:

```suggestion
				expect(
					registry
						.select( CORE_UI )
						.getValue(
							UI_KEY_SHOW_RRM_PUBLICATION_APPROVED_NOTIFICATION
						)
				).toBe( undefined );
```